Add ActivityRegistry signals for ESModule construction

Dr15Jones opened the pull request:
#### PR description:

- Added signals for pre and post ESModule construction
- extended Tracer output to include those signals
- extended file generated by Tracer to include those signals
- extended the web-based viewer to show the ESModule construction timing

#### PR validation:

Framework test all pass. Running a simple hand written configuration showed the output.

resolves https://github.com/cms-sw/framework-team/issues/1661
resolves https://github.com/cms-sw/framework-team/issues/1662
resolves https://github.com/cms-sw/framework-team/issues/1663
resolves https://github.com/cms-sw/framework-team/issues/1664

Dr15Jones committed:
Add ActivityRegistry signals for ESModule construction

- Added signals for pre and post ESModule construction
- extended Tracer output to include those signals

Dr15Jones commented:
please test

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49380/46803

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49380/46803/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49380/46803/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49380/46809

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49380/46813

cmsbuild commented:
Pull request #49380 was updated. @Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please check and sign again.

Dr15Jones commented on FWCore/Framework/interface/ComponentConstructionSentry.h:
```diff
@@ -0,0 +1,35 @@
+
+    struct ComponentConstructionSentry {
+      ComponentConstructionSentry(EventSetupProvider const& iProvider, ComponentDescription const& iDescription);
+      ~ComponentConstructionSentry();
```
After a short discussion, given the reference based member function already forbid such usage, we decided not to add them.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49380/46816

cmsbuild commented:
Pull request #49380 was updated. @Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please check and sign again.

Dr15Jones committed:
fix test_tracer_file.sh

The test did not properly handle failing cases and some grep commands were ambiguous

Dr15Jones committed:
Add ES module construction to tracer failes

- edmTracerCompactLogViewer can handle  ES module construction
- extend test configuration to include ES modules

Dr15Jones committed:
Need to call succeeded

Dr15Jones commented:
please test

Dr15Jones committed:
forward declare as struct for clang

makortel commented:
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/47071

makortel commented:
+core
